conans/test/integration/command_v2/list_package_ids_test.py

```diff
@@ -0,0 +1,180 @@
+
+        self.client.run("list package-ids --remote remote1 --cache", assert_error=True)
+        assert "error: the following arguments are required: reference" in self.client.out
+
+    def test_remote_and_all_remotes_are_mutually_exclusive(self):
+        self._add_remote("remote1")
+
+        self.client.run("list package-ids --all-remotes --remote remote1 package/1.0", assert_error=True)
+        assert "error: argument -r/--remote: not allowed with argument -a/--all-remotes" in self.client.out
+
+
+class TestListRecipesFromRemotes(TestListPackageIdsBase):
+    def test_by_default_search_only_in_cache(self):
+        self._add_remote("remote1")
+        self._add_remote("remote2")
+
+        expected_output = ("Local Cache:\n"
+                           "  There are no matching recipes\n")
+
+        self.client.run("list package-ids whatever/1.0")
+        assert expected_output == self.client.out
+
+    def test_search_no_matching_recipes(self):
+        self._add_remote("remote1")
+        self._add_remote("remote2")
+
+        expected_output = ("Local Cache:\n"
+                           "  There are no matching recipes\n"
+                           "remote1:\n"
+                           "  There are no matching recipes\n"
+                           "remote2:\n"
+                           "  There are no matching recipes\n")
+
+        self.client.run("list package-ids -c -a whatever/1.0")
+        assert expected_output == self.client.out
+
+    def test_fail_if_no_configured_remotes(self):
+        self.client.run("list package-ids -a whatever/1.0", assert_error=True)
+        assert "ERROR: The remotes registry is empty" in self.client.out
+
+    def test_search_disabled_remote(self):
+        self._add_remote("remote1")
+        self.client.run("remote disable remote1")
+        self.client.run("list package-ids -r remote1 whatever/1.0", assert_error=True)
+        assert "ERROR: Remote 'remote1' is disabled" in self.client.out
+
+
+class TestRemotes(TestListPackageIdsBase):
+    def test_search_with_full_reference(self):
+        remote_name = "remote1"
+        recipe_name = "test_recipe/1.0.0@user/channel"
+        self._add_remote(remote_name)
+        self._upload_recipe(remote_name, recipe_name)
+
+        self.client.run("list package-ids -r remote1 test_recipe/1.0.0@user/channel")
+
+        expected_output = (
+            r"remote1:\n"
+            r"  {}#.*\n".format(recipe_name)
+        )
+
+        assert bool(re.match(expected_output, str(self.client.out), re.MULTILINE))
+
+    def test_search_without_user_and_channel(self):
+        remote_name = "remote1"
+        recipe_name = "test_recipe/1.0.0@user/channel"
+        self._add_remote(remote_name)
+        self._upload_recipe(remote_name, recipe_name)
+
+        self.client.run("list package-ids -r remote1 test_recipe/1.0.0")
+
+        expected_output = (
+            "remote1:\n"
+            "  There are no matching recipes\n"
```

Comment:
I cannot find a test with recipe but without binaries: What is the output in that case?